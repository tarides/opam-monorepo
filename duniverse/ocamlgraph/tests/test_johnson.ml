
(* Test file for Johnson *)

open Graph

module Int = struct
  type t = int
  let compare = compare
  let hash = Hashtbl.hash
  let equal = (=)
  let default = 0
  end


module G = Imperative.Digraph.ConcreteLabeled(Int)(Int)


module W = struct
  type edge = G.E.t
  type t = int
  let weight e = G.E.label e
  let zero = 0
  let add = (+)
  let sub = (-)
  let compare = compare
  end

module J = Path.Johnson(G)(W)

let g = G.create ()

let () =
  G.add_edge_e g (G.E.create 1 3  2);
  G.add_edge_e g (G.E.create 1 (-4)  5);
  G.add_edge_e g (G.E.create 1 8  3);
  G.add_edge_e g (G.E.create 2 7  5);
  G.add_edge_e g (G.E.create  2 1  4);
  G.add_edge_e g (G.E.create  3 4  2);
  G.add_edge_e g (G.E.create  4 (-5)  3);
  G.add_edge_e g (G.E.create  4 2  1);
  G.add_edge_e g (G.E.create  5 6  4)

let () = let test = J.all_pairs_shortest_paths g in
	 J.HVV.iter (fun (v, u) d -> Printf.printf "[%d -> %d : %d]\n" v u d) test
