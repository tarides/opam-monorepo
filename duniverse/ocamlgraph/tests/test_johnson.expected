[2 -> 5 : -1]
[5 -> 1 : 8]
[2 -> 4 : 1]
[2 -> 1 : 3]
[3 -> 2 : 4]
[3 -> 5 : 3]
[5 -> 2 : 5]
[1 -> 1 : 0]
[2 -> 3 : -4]
[1 -> 4 : 2]
[5 -> 5 : 0]
[4 -> 4 : 0]
[5 -> 3 : 1]
[1 -> 5 : -4]
[4 -> 5 : -2]
[1 -> 2 : 1]
[2 -> 2 : 0]
[4 -> 3 : -5]
[3 -> 1 : 7]
[3 -> 4 : 5]
[4 -> 2 : -1]
[4 -> 1 : 2]
[1 -> 3 : -3]
[5 -> 4 : 6]
[3 -> 3 : 0]
